Quantum-Control-Applications/Superconducting/Multi Flux Tunable transmon/Use Case 1 - Two qubit gate optimization with cryoscope/readme.md

```diff
@@ -0,0 +1,188 @@
+* The second important quantity to calibrate is the $\pi/2$ pulse. This can be done using a simple Rabi sequence where both the amplitude and frequency are scanned while the flux bias is set to the qubit flux insensitive point found previously.
+* (optional) If the qubit pulses are relatively short (< 100ns), then the use of optimal DRAG waveforms can enhance the fidelity of the gates [[3]](#3). 
+* The last calibration to perform is the qubit detuning versus flux bias. This will be used to validate the qubit flux insensitive point and the avoided level crossing, as well as the cryoscope measurement. The sequence can be found in the [Cryoscope use-case](#2).
+
+## 2. The configuration file
+
+### 3.1 The elements
+The configuration consists of 3 elements:
+* `qubit0` and `qubit1` send control pulses to each transmon
+* `resonator0` and `resonator1` send pump pulses and measures the reflected signal for each qubit
+* `flux_line0` and `flux_line1` send control pulses and DC voltage to the flux line of each qubit
+
+
+### 3.2 The operations, pulse and waveforms
+The qubits elements `qubit0` and `qubit1` have several operations defined that correspond to X/2, Y/2 and X gates.
+
+Since a $\pi$-pulse is 100ns long for `qubit1`, it uses a simple Gaussian envelop.
+On the other hand, since a $\pi$-pulse is 16ns long for `qubit0`, an optimized DRAG waveform [[3]](#3) is used to generate its gates.
+
+
+The `resonator` has just one readout operation defined as a constant amplitude pulse
+
+The flux line elements `flux_line0` and `flux_line1` have a single biasing operation consisting of a square pulse added on top of a DC offset set to the flux soft spot.
+
+## 3. The QUA program and data processing
+The QUA program consists of three distinct parts:
+1. Measuring the flux line step response by implementing cryoscope.
+2. From this step response, derive the IIR and FIR filters to pre-distort the flux pulse.
+3. Characterize the corrections by implementing a two qubit chevron experiment between |11> and |02> (CZ gate).
+
+### 3.1 [Cryoscope](cryoscope.py)
+
+ The idea of the method is to measure the step response of the flux line using the qubit phase measured in a Ramsey-like experiment [[1]](#1). 
+ The possibly distorted response can then be pre-corrected by designing suitable IIR and FIR filters.
+
+This experiment and implementation in QUA has already been documented in the following [use-case](#2).
+The main difference here is that a constant amplitude flux pulse has been used to measure the step response of the flux line.
+
+
+### 3.2 [IIR and FIR filter implementation](cryoscope.py)
+
+The method used to design the filters is to fit the flux step response with one or several exponential decay functions 
+and use analytical laws to derive the corresponding FIR and IIR taps. 

```

Comment:
analytical formulas or functions? laws sounds like gravity

Reply:
Where do they come from by the way? Tomer?

Reply:
Tomer derive them but it comes from some math that Uri taught Tomer or pointed Tomer to

Reply:
It is kind of the class from university where you learn digital filters if you are ELE

Reply:
Ok cool :)